bbq/vm/vm.go

```diff
@@ -1539,6 +1539,41 @@ func opRemoveTypeIndex(vm *VM, ins opcode.InstructionRemoveTypeIndex) {
 	}
 }
 
+func opGetFieldLocal(vm *VM, ins opcode.InstructionGetFieldLocal) {
+	// get local
+	localIndex := ins.Local
+	absoluteIndex := vm.callFrame.localsOffset + localIndex
+	local := vm.locals[absoluteIndex]
+
+	// Some local variables can be implicit references. e.g: receiver of a bound function.
+	// TODO: maybe perform this check only if `localIndex == 0`?
+	if implicitReference, ok := local.(ImplicitReferenceValue); ok {
+		local = implicitReference.ReferencedValue(vm.context)
+	}
+
+	// get field
+	memberAccessibleValue := local.(interpreter.MemberAccessibleValue)
+
+	checkMemberAccessTargetType(
+		vm,
+		ins.AccessedType,
+		memberAccessibleValue,
+	)
+
+	// VM assumes the field name is always a string.
+	fieldNameIndex := ins.FieldName
+	fieldName := getRawStringConstant(vm, fieldNameIndex)
+
+	fieldValue := memberAccessibleValue.GetMember(vm.context, fieldName)
+	if fieldValue == nil {
+		panic(&interpreter.UseBeforeInitializationError{
+			Name: fieldName,
+		})
+	}
+
+	vm.push(fieldValue)
```

Comment:
Would it be possible to reuse the existing components of the instructions, i.e. `opGetLocal` and `opGetField`? We might have to refactor parts of those to make their logic reusable (i.e. avoid a temporary push and pop to the stack)